### PR TITLE
SDIT-1149 Prevent updates to Activities for prisons using DPS Activities

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/service/ReleaseAPrisonerIntegrationTest.java [hunk 1]
```diff
@@ -146,6 +163,28 @@ public void deactivateAllProgramProfiles() {
         assertThat(activeProgramProfiles).isZero();
     }
 
+    @Test
+    public void dontDeactivateAllProgramProfiles() {
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/hmpps/prison/service/ReleaseAPrisonerIntegrationTest.java:167]: Test copied from an existing test but turns on the feature switch for DPS Activities first.

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/ServiceAgencySwitchesResource.kt [hunk 2]
```diff
@@ -55,13 +53,12 @@ class ServiceAgencySwitchesResource(private val service: ServiceAgencySwitchesSe
   @PreAuthorize("hasRole('SERVICE_AGENCY_SWITCHES')")
   @PostMapping("/{serviceCode}/prison/{prisonId}")
   @ResponseStatus(HttpStatus.CREATED)
-  @Hidden // TODO SDIT-797 Remove annotation once tables EXTERNAL_SERVICES  and SERVICE_AGENCY_SWITCHES are live
   fun addServicePrison(
     @PathVariable
-    @Parameter(name = "The code of the service from the EXTERNAL_SERVICES table")
+    @Parameter(description = "The code of the service from the EXTERNAL_SERVICES table")
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/ServiceAgencySwitchesResource.kt:58]: This caused the Open API tests to fail - not sure why they just started failing but clearly needed fixing.

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java [hunk 3]
```diff
@@ -136,7 +137,7 @@ public InmateDetail releasePrisoner(final String prisonerIdentifier, final Reque
 
         updatePayPeriods(booking.getBookingId(), releaseDateTime.toLocalDate());
 
-        deactivateEvents(booking.getBookingId());
+        deactivateEvents(booking.getBookingId(), booking.getLocation().getId());
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java:140]: double checking, this can't be null?

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**